pysat/_instrument.py

```diff
@@ -458,16 +464,21 @@ def __getitem_xarray__(self, key):
             inst[datetime1:datetime1, 'name1':'name2']
 
         """
-        if 'time' not in self.data:
+        if 'Epoch' in self.data.indexes:
+            epoch_name = 'Epoch'
+        elif 'time' in self.data.indexes:
+            epoch_name = 'time'
+        else:
             return xr.Dataset(None)
+
         if isinstance(key, tuple):
             if len(key) == 2:
                 # support slicing time, variable name
                 try:
-                    return self.data.isel(time=key[0])[key[1]]
+                    return self.data.isel(indexers={epoch_name: key[0]})[key[1]]
                 except:
                     try:
-                        return self.data.sel(time=key[0])[key[1]]
+                        return self.data.sel(indexers={epoch_name: key[0]})[key[1]]
                     except TypeError: # construct dataset from names
                         return self.data[self.variables[key[1]]]
```

Comment:
Should this be if/else instead of try/except?

Reply:
The use of duck typing here dates back to #125. See the changes in _instrument.py here: https://github.com/pysat/pysat/pull/125/commits/124a87eecd85a7716bd628bb0c3494ac76c29bef#

Basically, we support a lot of types of input data, which requires a lot of nested if/elif statements that boil down to the same thing.  Updating these with (TypeError, KeyError) to at least get rid of the bare except statements.